### PR TITLE
3 define api spec

### LINKED ISSUE #3: Define API spec
We will need to coordinate with other teams to determine which routes and objects our service should offer/support to create and validate schedules.

### PR DESCRIPTION
**Pull request by c3n0te**
Pulled API spec into local filesystem as a yaml file and used fastapi-code-generator package (pre-stable-release tool) to generate pydantic models from api object schemas. Had to make minor changes to the spec as defined in swaggerhub due to some unforeseen difficulties with how fastapi/python converts container types. Went back and made modifications in swaggerhub but will need to validate again to ensure compliance with the spec. After the pydantic models and fastapi endpoints were generated, the endpoints were tweaked manually and integrated into the ray serve framework.

### CODE REVIEW

**Comment by c3n0te** [comment]
Lol get blocked nerd...

Guess our github actions are working well, can someone explain to me how the linting was configured in lab tomorrow please and thank you?

File: api/src/main.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from typing import Union
+from hypergraph import HyperGraph
+from lib import preprocess
+
+from models import (
+    Error,
+    InputData,
+    IsValidSchedule,
+    Schedule,
+    Success,
```
Comment (seiyaterada) [inline api/src/main.py]: Looks like this model isn't being used anywhere

**Review by seiyaterada — APPROVED** [review]
Wasn't able to test it locally, but just from looking at the code I think it looks great! Just added a comment about the Success model not being used which I mentioned in the lab.

File: api/src/main.py (hunk 1 above)
Comment (c3n0te) [reply api/src/main.py]: Just removed Success and Error types, good catch

**Comment by c3n0te** [comment]
pull the lever kronk

**Pull request merged**